src/ert/job_queue/job_queue_node.py

```diff
@@ -258,43 +303,52 @@ def _job_monitor(
                     self.run_done_callback()
 
             # refresh cached status after running the callback
-            current_status = self.refresh_status(driver)
-            if current_status == JobStatusType.JOB_QUEUE_SUCCESS:
-                pass
-            elif current_status == JobStatusType.JOB_QUEUE_EXIT:
+            current_status = self._poll_queue_status(driver)
+
+            if current_status in self.DONE_STATES:
+                self._transition_status(ThreadStatus.DONE, current_status)
+            elif current_status in self.RESUBMIT_STATES:
                 if self.submit_attempt < max_submit:
                     logger.warning(
                         f"Realization: {self.callback_arguments[0].iens} "
                         f"failed with: {self._status_msg}, resubmitting"
                     )
-                    self._set_thread_status(ThreadStatus.READY)
-                    return
+                    self._transition_status(ThreadStatus.READY, current_status)
                 else:
-                    logger.error(
-                        f"Realization: {self.callback_arguments[0].iens} "
+                    self._transition_to_failure(
+                        message=f"Realization: {self.callback_arguments[0].iens} "
                         f"failed after reaching max submit with: {self._status_msg}"
```

Comment:
how do you feel about adding the number max submit in the log here?
```suggestion
                        f"failed after reaching max submit {max_submit} with: {self._status_msg}"
```